src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -1023,33 +1023,36 @@ namespace AirLoopHVACDOAS {
-                } // else { // What about other cases?
-                state.dataAirLoopHVACDOAS->SummerDesignDayFlag = false;
+            if (desDayInput.DayType == summerDesignDayTypeIndex) {
+                // keep design day humiditiy ratio from the same peak temperature design day (i.e., don't use a humrat from a different day)
+                if (desDayInput.MaxDryBulb > this->SizingCoolOATemp) {
+                    this->SizingCoolOATemp = desDayInput.MaxDryBulb;
+                    if (desDayInput.HumIndType == WeatherManager::DDHumIndType::WetBulb) { // wet bulb temperature
+                        this->SizingCoolOAHumRat = Psychrometrics::PsyWFnTdbTwbPb(
+                            state, this->SizingCoolOATemp, desDayInput.HumIndValue, DataEnvironment::StdPressureSeaLevel);
+                    } else if (desDayInput.HumIndType == WeatherManager::DDHumIndType::DewPoint) { // dewpoint
+                        this->SizingCoolOAHumRat = Psychrometrics::PsyWFnTdpPb(state, desDayInput.HumIndValue, DataEnvironment::StdPressureSeaLevel);
+                    } else if (desDayInput.HumIndType == WeatherManager::DDHumIndType::HumRatio) {
+                        this->SizingCoolOAHumRat = desDayInput.HumIndValue;
+                    } // else { // What about other cases?
+                }
             }
             // Winter design day
-            if (state.dataWeatherManager->DesDayInput(i).DayType == winterDesignDayTypeIndex && state.dataAirLoopHVACDOAS->WinterDesignDayFlag) {
-                this->HeatOutTemp = state.dataWeatherManager->DesDayInput(i).MaxDryBulb;
-                if (state.dataWeatherManager->DesDayInput(i).HumIndType == WeatherManager::DDHumIndType::WetBulb) { // wet bulb temperature
-                    this->HeatOutHumRat = Psychrometrics::PsyWFnTdbTwbPb(
-                        state, this->HeatOutTemp, state.dataWeatherManager->DesDayInput(i).HumIndValue, DataEnvironment::StdPressureSeaLevel);
-                } else if (state.dataWeatherManager->DesDayInput(i).HumIndType == WeatherManager::DDHumIndType::DewPoint) { // dewpoint
-                    this->HeatOutHumRat = Psychrometrics::PsyWFnTdpPb(
-                        state, state.dataWeatherManager->DesDayInput(i).HumIndValue, DataEnvironment::StdPressureSeaLevel);
-                } else if (state.dataWeatherManager->DesDayInput(i).HumIndType == WeatherManager::DDHumIndType::HumRatio) {
-                    this->HeatOutHumRat = state.dataWeatherManager->DesDayInput(i).HumIndValue;
-                } // else { // What about other cases?
-                state.dataAirLoopHVACDOAS->WinterDesignDayFlag = false;
+            if (desDayInput.DayType == winterDesignDayTypeIndex) {
+                // keep design day humiditiy ratio from the same peak temperature design day (i.e., don't use a humrat from a different day)
+                if (desDayInput.MaxDryBulb < this->HeatOutTemp) {
+                    this->HeatOutTemp = desDayInput.MaxDryBulb;
```

Comment:
Should be `desDayInput.MaxDrybulb - desDayInput.DailyDBRange` in both lines.

Reply:
I think I'll make this change now since it is the correct math, and will cause diffs but that is expected.